datahub-actions/src/datahub_actions/event/event_registry.py

```diff
@@ -60,7 +60,19 @@ def from_class(cls, clazz: EntityChangeEventClass) -> "EntityChangeEvent":
     @classmethod
     def from_json(cls, json_str: str) -> "EntityChangeEvent":
         json_obj = json.loads(json_str)
-        return cls.from_obj(json_obj, True)
+
+        # Remove parameters from json_obj and add it later to _inner_dict, this hack exists because of the way EntityChangeLogClass does not support "AnyRecord"
+        parameters = json_obj.pop("parameters", None)
+
+        event = cls.from_class(cls.from_obj(json_obj))
+
+        # Hack: Since parameters is an "AnyRecord" (arbitrary json) we have to insert into the underlying map directly
+        # to avoid validation at object creation time. This means the reader is responsible to understand the serialized JSON format, which
+        # is simply PDL serialized to JSON.
+        if parameters:
+            event._inner_dict["__parameters_json"] = parameters
+
+        return event
 
     def as_json(self) -> str:
         json_obj = self.to_obj()
```

Comment:
I guess the downside to keeping this in __parameters_json is that it's not easy to access for the consumer :(

Reply:
Without first serializing to json of course